components/mobile-plugins/android-plugin/org.wso2.carbon.device.mgt.mobile.android.emulator/src/main/java/org.carbon.android.emulator/TryIt.java

```
@@ -0,0 +1,828 @@
+        System.exit(1);
+    }
+
+    /**
+     * This method check for the android agent in the specified AVD and installs it if not available.
+     *
+     * @return package name and act name.
+     */
+    private String[] checkForAgent() {
+        String pkg = null;
+        String activity = null;
+        String readLine;
+        BufferedReader reader = null;
+        String apkFileLocation = workingDirectory + Constants.APK_LOCATION;
+        String aaptLocation = androidSdkHome + File.separator + "build-tools" + File.separator
+                + System.getProperty(Constants.BUILD_TOOLS_VERSION) + File.separator + "aapt";
+        if (osSuffix.equals(Constants.WINDOWS_OS)) {
+            aaptLocation += Constants.WINDOWS_EXTENSION_EXE;
+        }
+        setExecutePermission(aaptLocation);
+        ProcessBuilder badgingApkFileProcessBuilder = new ProcessBuilder(aaptLocation, "d", "badging",
+                apkFileLocation);
+        try {
+            Process badgingApkFileProcess = badgingApkFileProcessBuilder.start();
+            reader = new BufferedReader(new InputStreamReader(badgingApkFileProcess.getInputStream(),
+                    StandardCharsets.UTF_8));
+            while ((readLine = reader.readLine()) != null) {
+                if (readLine.contains("package")) {
+                    Pattern pattern = Pattern.compile("'(.*?)'");
+                    Matcher matcher = pattern.matcher(readLine);
+                    if (matcher.find()) {
+                        pkg = matcher.group(1);
+                    }
+                }
+                if (readLine.contains("launchable-activity")) {
+                    Pattern pattern = Pattern.compile("'(.*?)'");
+                    Matcher matcher = pattern.matcher(readLine);
+                    if (matcher.find()) {
+                        activity = matcher.group(1);
+                    }
+                }
+            }
+        } catch (IOException ignored) {
+            //
```

Can we really ignore this exception? It'll be occurring when there is an issue in getting the process logs right. In that case, better atleast log the WARN log with the error message. No need the stacktrace.